amazon/ion/ioncmodule.c

```diff
@@ -1047,7 +1103,7 @@ static iERR ionc_read_timestamp(hREADER hreader, PyObject** timestamp_out) {
                     decContextClearStatus(&dec_context, DEC_Inexact);
                 }
 
-                char* dec_num = malloc(fractional_precision + 2);
+                char* dec_num = malloc(fractional_precision + 6); // TODO: decQuadToString call below seems to overflow the allocated memory
```

Comment:
Valgrind revealed some illegal memory access made by the 3rd party code where it tried to write beyond the number of bytes we allocated here. I've bumped up the padding a bit, but still not sure if this is the right way

Reply:
Dumped some info from my environment [here](https://gist.github.com/alimg/632440698bcf8949f8b6fc8552de9670) but I must admit I don't know what could be platform dependent here.

Is there a reason not to use [DECQUAD_String](https://github.com/amzn/ion-c/blob/a541a810b8e0471262f7bfda04283859a07b7d10/decNumber/example7.c#L18) here? I've updated this to match the other examples used in the ion-c code base, but I hope I didn't remove any specific memory saving trick put in place before.

Reply:
Yeah I agree, using `DECQUAD_String` is a good idea. Change `+2` to `+6` is not a big deal too. I'm just curious dump what data causes the illegal memory access.